app/code/core/Maho/Captcha/Helper/Data.php

```diff
@@ -69,14 +69,15 @@ public function getWidgetAttributes(): Varien_Object
 
     public function createChallenge(?array $options = null): Altcha\Challenge
     {
-        $options = new Altcha\ChallengeOptions([
-            'algorithm' => Altcha\Algorithm::SHA512,
-            'saltLength' => 32,
-            'expires' => (new DateTime())->modify('+' . self::CHALLENGE_EXPIRATION . ' seconds'),
-            'hmacKey' => $this->getHmacKey(),
-            ...($options ?? []),
-        ]);
-        return Altcha\Altcha::createChallenge($options);
+        $options = new Altcha\ChallengeOptions(
+            Altcha\Hasher\Algorithm::SHA512,
+            Altcha\ChallengeOptions::DEFAULT_MAX_NUMBER,
+            (new DateTime())->modify('+' . self::CHALLENGE_EXPIRATION . ' seconds'),
+            $options ?? [],
+            32,
+        );
```

Comment:
```suggestion
        $options = new Altcha\ChallengeOptions(
            algorithm: Altcha\Hasher\Algorithm::SHA512,
            maxNumber: Altcha\ChallengeOptions::DEFAULT_MAX_NUMBER,
            expires: (new DateTime())->modify('+' . self::CHALLENGE_EXPIRATION . ' seconds'),
            params: $options ?? [],
            saltLength: 32,
        );
```

We could use named parameters, is it better?

Reply:
phpstorm already shows the argument names so in this case I wouldn't add the names. it would allow us to remove the maxNumber parameter but... who cares for this specific thing 😅